crud/src/main/java/com/redhat/lightblue/rest/crud/hystrix/GenerateCommand.java

```diff
@@ -0,0 +1,131 @@
+
+import com.redhat.lightblue.config.LightblueFactory;
+
+import com.redhat.lightblue.query.FieldProjection;
+
+import com.redhat.lightblue.crud.Factory;
+import com.redhat.lightblue.crud.CrudConstants;
+import com.redhat.lightblue.crud.valuegenerators.GeneratedFields;
+
+import com.redhat.lightblue.mediator.DefaultMetadataResolver;
+
+import com.redhat.lightblue.metadata.Metadata;
+import com.redhat.lightblue.metadata.EntityMetadata;
+import com.redhat.lightblue.metadata.FieldTreeNode;
+import com.redhat.lightblue.metadata.SimpleField;
+
+import com.redhat.lightblue.util.Error;
+import com.redhat.lightblue.util.Path;
+import com.redhat.lightblue.rest.CallStatus;
+import com.redhat.lightblue.rest.RestConfiguration;
+import com.redhat.lightblue.rest.crud.RestCrudConstants;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GenerateCommand extends AbstractRestCommand {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GenerateCommand.class);
+
+    private final String entity;
+    private final String version;
+    private final String field;
+    private final int n;
+
+    public GenerateCommand(String clientKey, String entity, String version, String field,int n) {
+        super(GenerateCommand.class, clientKey,null);
+        this.entity = entity;
+        this.version = version;
+        this.field = field;
+        this.n = n<=0?1:n;
+    }
+    
+    @Override
+    protected CallStatus run() {
+        LOGGER.debug("run: entity={}, version={}", entity, version);
+        Error.reset();
+        Error.push("rest");
+        Error.push(getClass().getSimpleName());
+        Error.push(entity);
+        try {
+            LightblueFactory lbf=RestConfiguration.getFactory();
+            Factory factory=lbf.getFactory();
+            Metadata md=lbf.getMetadata();
+            DefaultMetadataResolver mdResolver=new DefaultMetadataResolver(md);
+            Path fieldPath=new Path(field);
+            mdResolver.initialize(entity,version,null,new FieldProjection(fieldPath,true,false));
+            HttpServletRequest sreq=ResteasyProviderFactory.getContextData(HttpServletRequest.class);
+            Set<String> userRoles=new HashSet<>();
+            if(sreq!=null) {
+                for(String role:mdResolver.getMetadataRoles()) {
+                    if(sreq.isUserInRole(role))
+                        userRoles.add(role);
+                }
+            }
+            LOGGER.debug("user roles:{}",userRoles);
+            EntityMetadata emd=mdResolver.getCompositeMetadata();
+            if(emd.getAccess().getFind().hasAccess(userRoles)||
+               emd.getAccess().getInsert().hasAccess(userRoles)||
+               emd.getAccess().getUpdate().hasAccess(userRoles)) {
+                LOGGER.debug("User has access, looking up {}",fieldPath);
+                FieldTreeNode ftn=emd.resolve(fieldPath);
+                LOGGER.debug("Field={}",ftn);
+                SimpleField fld=(SimpleField)ftn;
```

Comment:
What if the field given is not a simple field?  Maybe the request is messed up and path is to an object or array.  Would be better to handle here than bubble up class cast exception.